cpp/src/groupby/groupby.cu

```diff
@@ -43,6 +43,19 @@
 
 namespace cudf {
 namespace groupby {
+namespace {
+/**
+ * @brief Helper functor to check if a specified type supports equality comparisons.
+ */
+struct equality_comparable_functor {
+  template <typename T>
+  bool operator()() const
+  {
+    return cudf::is_equality_comparable<T, T>();
+  }
+};
+}  // namespace
```

Comment:
Make a "runtime" trait for `data_type` that wraps this with the `type_dispacher` in `traits.hpp`.

Reply:
See https://github.com/rapidsai/cudf/blob/0cf2c5ff32daca70e58c2acfc78ca28fc333ba18/cpp/include/cudf/utilities/traits.hpp#L158-L180 for example

Reply:
Ah yes, this check is very useful for general usage too.

Reply:
Hey, @jrhemstad. Sorry, I haven't grokked your advice here. Please ELI5?

Reply:
Oh, I see... Make an `is_equality_comparable(column_view, column_view)` available in `traits.hpp`, and use that here?

Reply:
Not quite. `is_equality_comparable(data_type)` is what I meant.

Reply:
I should really look into why GitHub notifications arrive so late.)

I tried writing a binary version of the same (with differing `lhs` and `rhs` type). That would require double dispatch, even for a single type.
(That was a long walk down a winding road to a cafe that was closed. :p) I'll have the unary version up shortly.

Reply:
This is sorted out. I've left the binary version of `is_equality_comparable()` out of this PR, since it's not necessary for this change.